fontir/src/orchestration.rs

```diff
@@ -30,6 +30,52 @@ bitflags! {
     }
 }
 
+/// Generates fn $getter_name(&self) -> Arc<$value_type>.
+///
+/// Assumes we are in an impl block for a Context and that
+/// self.$lock_name is an Arc<RwLock<Option<Arc<$ir_type>>>>
+///
+/// <https://veykril.github.io/tlborm/decl-macros/minutiae/fragment-specifiers.html>
+#[macro_export]
+macro_rules! get_from_context {
+    ($getter_name:ident, $lock_name:ident, $value_type:ty, $id:expr, $restore_fn:ident) => {
+        pub fn $getter_name(&self) -> Arc<$value_type> {
+            let id = $id;
```

Comment:
I wonder if we would benefit from having some simple trait that associates an id type with the resource type that it identifies? Like,


```rust
trait ContextItem {
    fn token(&self) -> WorkId;
}
```

basically what I'm wondering is if there's a way to express this in the type system such that you actually end up with a single pair of methods on the context that look like,

```rust
fn get<T: ContextItem>(&self) -> Arc<T> { .. }
fn set<T: ContextItem>(&self, item: T) { .. }
```

I'm not sure how possible this is, but it's something I would be thinking about if I were writing this.

Reply:
I like the idea but I want to push towards #138 so deferring to https://github.com/googlefonts/fontmake-rs/issues/150.

Reply:
I like it but I want to tackle it in it's own PR.